src/types/any.rs

```diff
@@ -1862,6 +1871,30 @@ impl<'py> PyAnyMethods<'py> for Bound<'py, PyAny> {
         inner(self, other.to_object(py).into_bound(py), compare_op)
     }
 
+    fn neg(&self) -> PyResult<Bound<'py, PyAny>> {
+        fn inner<'py>(any: &Bound<'py, PyAny>) -> PyResult<Bound<'py, PyAny>> {
+            unsafe { ffi::PyNumber_Negative(any.as_ptr()).assume_owned_or_err(any.py()) }
+        }
+
+        inner(self)
```

Comment:
Because these methods aren't generic they don't gain any value from the inner function pattern, so we can simplify:

```suggestion
        unsafe { ffi::PyNumber_Negative(self.as_ptr()).assume_owned_or_err(self.py()) }
```

Same for the others below.

The `inner` function is useful when we have one or more generic arguments like `O: ToPyObject`. By having the `inner` function and calling it with the result of calling `.to_object()` we make the inner code not generic, which helps keep overall size of things down.

Reply:
I see, got it